### ISSUE TITLE
no-free-form: test could be a bit more precise

### ISSUE THREAD

**Issue by Alphix** [ansible/ansible-lint#2597]
##### Summary

The `no-free-form` test currently triggers when a single `=` is found in an action value. I'm guessing this is the relevant test: 

https://github.com/ansible/ansible-lint/blob/3a9e2904b6a70435f3dbd50e37387dcce86a5da0/src/ansiblelint/rules/no_free_form.py#L61

The test could be a bit more specific and exclude e.g. `==`.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible [core 2.13.4] (Debian Unstable)
ansible-lint 6.8.2.dev3 using ansible 2.13.4 (GitHub checkout from today using pip)
```

- ansible installation method: OS package (Debian Unstable)
- ansible-lint installation method: source (`pip3 install git+https://github.com/ansible-community/ansible-lint.git`)

##### OS / ENVIRONMENT

Debian Unstable, but not relevant

##### STEPS TO REPRODUCE

Here's a couple of example false positives from DebOps:

https://github.com/debops/debops/blob/24a41214c212b79b3c166739b78a0c22e1b24ecc/ansible/roles/apparmor/tasks/main.yml#L77-L90

(triggered by the `=` in the regex....could perhaps be fixed by checking if the left-hand-side of the `=` is a valid Ansible module parameter name)

https://github.com/debops/debops/blob/24a41214c212b79b3c166739b78a0c22e1b24ecc/ansible/roles/cryptsetup/tasks/manage_devices.yml#L41-L53

(triggered by the `==`)

https://github.com/debops/debops/blob/24a41214c212b79b3c166739b78a0c22e1b24ecc/ansible/roles/etesync/tasks/main.yml#L162-L181

(triggered by the `>=` in the comment to the shell invocation)

##### Desired Behavior

Things like `==`, `<=`, `>=` shouldn't trigger the `no-free-form` warning.

Ideally, things like `ansible.builtin.shell: grep --quiet "flags\s*=.*complain"` also shouldn't trigger the warning (but I guess this one might be more difficult).

##### Actual Behavior

Warnings are generated for things which are not actually in free-form syntax.

**Comment by ssbarnea** [comment]
Fixed by #2596 we will to a release soon.
